src/Mod/Draft/draftobjects/patharray.py

```diff
@@ -26,67 +26,92 @@
 # \ingroup DRAFT
 # \brief This module provides the object code for the Draft PathArray object.
 
-from PySide.QtCore import QT_TRANSLATE_NOOP
+#from PySide.QtCore import _tr
 
 import FreeCAD as App
 import DraftVecUtils
 
 from draftutils.utils import get_param
+from draftutils.messages import _msg, _wrn
+from draftutils.translate import _tr, translate
 
 from draftobjects.draftlink import DraftLink
 
-
 class PathArray(DraftLink):
     """The Draft Path Array object"""
-
-    def __init__(self,obj):
+    def __init__(self, obj, use_link=False):
```

Comment:
The `use_link` parameter doesn't seem to be set by the function using this class, but it is actually set by the parent class `DraftLink`.

If an object is passed, it sets `use_link` to `True`, otherwise it sets it to `False`. This is the reason the `make_path_array` function has something like this `PathArray(None)`.

Being honest, I don't quite get how the `DraftLink` class works. I hope realthunder can add more documentation to it.

Reply:
Removed use_link parameter.